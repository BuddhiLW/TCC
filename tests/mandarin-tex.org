# #+TITLE: Aula 20;  弟二十课
#+STARTUP: latexpreview
#+OPTIONS: tex:t
#+LATEX_COMPILER: xelatex
#+LATEX_HEADER: \usepackage{xltxtra}
#+LATEX_HEADER: \setmainfont{Source Han Sans CN}
#+LATEX_HEADER: \usepackage{verbatim, fancyvrb}
#+LATEX_HEADER: \setmonofont{AR PL UMing CN}[Scale = MatchLowercase, UprightFont = * Bold, ItalicFont = * Bold Italic, BoldFont = * Black, BoldItalicFont = * Black Italic]

# * 第二十课
# #+AUTHOR:老市:张小. 学生: 小白色 佩德罗.
* Preamble
** Begin document with fontspec
#+begin_src latex
\documentclass{article}
\usepackage{fontspec} %Font package
#+end_src

** Declare font (in your system)
We are declaring the command \ch{}.
#+begin_src latex
\newfontfamily\ch[Mapping=tex-text]{HAN NOM A} 
%AR PL UMing CN : light font
%AR PL UKai TW MBE : drawn font (marked traces)
\DeclareTextFontCommand{\unifont}{\ch}
#+end_src

* Call the function inside the document
#+begin_src latex
\begin{document}
\ch{你好我说汉语}
\end{document}
#+end_src

* Note
To know the name of the fonts installed in the system which render Mandarin, we can instal =forfont=

#+begin_src shell
fontfor -t 语
#+end_src

It will open a webpage rendering and showing the fonts available in the system.

#+ATTR_HTML: :width 400
[[file:pic-selected-210714-1454-51.png][file:~/PP/LaTeX/TCC/tests/pic-selected-210714-1454-51.png]]
