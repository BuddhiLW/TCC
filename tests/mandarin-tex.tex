% Created 2021-07-14 Wed 17:30
% Intended LaTeX compiler: xelatex
\documentclass[11pt]{article}
\usepackage{amsmath}
\usepackage{xltxtra}

%% Global (changing main font and mono font which is used in verbatim)
\setmainfont{Source Han Sans CN} % Font for the hole text
\setmonofont{AR PL UMing CN} % Font for verbatim

%% Locally (setting local variables)
\newfontfamily\ch[Mapping=tex-text]{HAN NOM A} 
%AR PL UMing CN : light font
%AR PL UKai TW MBE : drawn font (marked traces)
\DeclareTextFontCommand{\unifont}{\ch}

\usepackage{verbatim, fancyvrb}
\date{\today}
\title{A little guide of Chinese Mandarin fonts in \LaTeX}
\begin{document}

\maketitle
% \tableofcontents

\section{Preamble}
\label{sec:org9153cc4}
\subsection{Begin document with fontspec}
\label{sec:org4c87d9b}
\begin{verbatim}
\documentclass{article}
\usepackage{fontspec} %Font package
\end{verbatim}

\subsection{Declare font (in your system)}
\label{sec:org3c49420}
We can declare the command ch,
\begin{verbatim}
%% Preamble %%
\newfontfamily\ch[Mapping=tex-text]{HAN NOM A} 
%AR PL UMing CN : light font
%AR PL UKai TW MBE : drawn font (marked traces)
\DeclareTextFontCommand{\unifont}{\ch}
\end{verbatim}

\section{Call the function inside the document}
\label{sec:org9f0b2b6}

Examples of use {\verb+{\ch{语}}+}:{\ch{语}}.

\begin{verbatim}
\begin{document}
{\ch{你好我说汉语}}
\end{document}
\end{verbatim}

 $=$ {\ch{你好我说汉语}}

\clearpage
\section{Note}
\label{sec:org643caa5}
To know the name of the fonts installed in the system which render Mandarin, we can instal \texttt{forfont}, written in Rust. Then, call it in the shell,

\begin{verbatim}
$ fontfor -p 语
\end{verbatim}

It will open a webpage rendering and showing the fonts available in the system.

{\includegraphics[width=.9\linewidth]{/home/buddhilw/PP/LaTeX/TCC/tests/pic-selected-210714-1454-51.png}}

Also, we can specify general fonts to be used in the hole document, also.

\begin{verbatim}
%% Preamble %%
\setmainfont{Source Han Sans CN} % Font for the hole text
\setmonofont{AR PL UMing CN} % Font for verbatim
\end{verbatim}
\end{document}