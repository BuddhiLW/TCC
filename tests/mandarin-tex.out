\BOOKMARK [1][-]{section.1}{\376\377\000P\000r\000e\000a\000m\000b\000l\000e}{}% 1
\BOOKMARK [2][-]{subsection.1.1}{\376\377\000B\000e\000g\000i\000n\000\040\000d\000o\000c\000u\000m\000e\000n\000t\000\040\000w\000i\000t\000h\000\040\000f\000o\000n\000t\000s\000p\000e\000c}{section.1}% 2
\BOOKMARK [2][-]{subsection.1.2}{\376\377\000D\000e\000c\000l\000a\000r\000e\000\040\000f\000o\000n\000t\000\040\000\050\000i\000n\000\040\000y\000o\000u\000r\000\040\000s\000y\000s\000t\000e\000m\000\051}{section.1}% 3
\BOOKMARK [1][-]{section.2}{\376\377\000C\000a\000l\000l\000\040\000t\000h\000e\000\040\000f\000u\000n\000c\000t\000i\000o\000n\000\040\000i\000n\000s\000i\000d\000e\000\040\000t\000h\000e\000\040\000d\000o\000c\000u\000m\000e\000n\000t}{}% 4
\BOOKMARK [1][-]{section.3}{\376\377\000N\000o\000t\000e}{}% 5
